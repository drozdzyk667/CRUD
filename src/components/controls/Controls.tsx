import React from 'react';
import Icon from 'Icons';
import { makeStyles, Box } from '@material-ui/core';
import { ControlsContainer, Button } from '../styled';
import { Colors, Names } from 'pages/recipes/Recipes.constants';

const useStyles = makeStyles({
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center'
  },
  text: {
    margin: '0 0.5em'
  }
});

interface ControlProps {
  handleOpenForm: (type: string) => void;
}

const Controls: React.FC<ControlProps> = ({ handleOpenForm }) => {
  const classes = useStyles();
  return (
    <ControlsContainer>
      <Button onClick={() => handleOpenForm(Names.add)} background={Colors.add}>
        <Box className={classes.container}>
          <span className={classes.text}>{Names.add}</span>
          <Icon name='Add' />
        </Box>
      </Button>
    </ControlsContainer>
  );
};

export default Controls;
