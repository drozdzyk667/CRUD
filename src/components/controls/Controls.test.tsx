import React from 'react';
import { shallow } from 'enzyme';
import Controls from './Controls';

describe('RecipeElement', () => {
  const clickFnOpen = jest.fn();

  it('check return elements', () => {
    const component = shallow(<Controls handleOpenForm={clickFnOpen} />);
    expect(component).toMatchSnapshot();
  });

  it('should render Add Recipe name', () => {
    const component = shallow(<Controls handleOpenForm={clickFnOpen} />);
    const btn = component.find({
      'data-testid': 'Add Recipe'
    });
    btn.first().simulate('click');
    expect(btn.first().props().children).toEqual('Add Recipe');
  });
  it('should render add button', () => {
    const component = shallow(<Controls handleOpenForm={clickFnOpen} />);
    const btn = component.find({
      'data-testid': 'add-button'
    });
    expect(btn.first().length).toBe(1);
  });
});
