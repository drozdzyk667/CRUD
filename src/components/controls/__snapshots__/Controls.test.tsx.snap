// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeElement check return elements 1`] = `
<styled.div>
  <styled.button
    background="#3db6f2"
    data-testid="add-button"
    onClick={[Function]}
  >
    <Styled(MuiBox)
      className="makeStyles-container-1"
    >
      <span
        className="makeStyles-text-2"
        data-testid="Add Recipe"
      >
        Add Recipe
      </span>
      <ForwardRef
        name="Add"
      />
    </Styled(MuiBox)>
  </styled.button>
</styled.div>
`;
